src/index.js

```diff
@@ -1,13 +1,126 @@
-export default function BaseballGame() {
-  this.play = function (computerInputNumbers, userInputNumbers) {
-    return "결과 값 String";
-  };
-}
-
-// export default class BaseballGame {
-//   play(computerInputNumbers, userInputNumbers) {
-//     return "결과 값 String";
-//   }
-// }
-
-new BaseballGame();
+export default class BaseballGame { 
+
+    constructor() {
+      this.computerNum = this.randomNumGenerator();
+      this.strike = 0;
+      this.ball = 0;
+      this.result = '';
+      console.log(this.computerNum);
+    }
+  
+    randomNumGenerator() {
+      let first = Math.floor(Math.random() * 9 + 1);
+      let second = Math.floor(Math.random() * 9 + 1);
+      while (first == second) {
+        second = Math.floor(Math.random() * 9 + 1);
+      }
+      let third = Math.floor(Math.random() * 9 + 1);
+      while (third == first || third == second) {
+        third = Math.floor(Math.random() * 9 + 1);
+      }
+      return (
+        third.toString() + second.toString() + first.toString()
+      );
+    }
+
+    checkUserInput(userInputNumbers) {
+        let userNumArray = userInputNumbers.split('');
+        if (userNumArray.length != 3) {
+            return false;
+        }
+        else if (isNaN(userInputNumbers) == true) {
+            return false;
+        }
+        let ret = true;
+        userNumArray.forEach(function(item, index) {
+            if (item == userNumArray[(index + 1) % 3] || item == userNumArray[(index + 2) % 3]) {
+                ret = false;
+            }
+        })
+        if (ret == false) {
+            return false;
+        }
+        return true;
+    }
+  
+    play(computerInputNumbers, userInputNumbers) {
+      let computerNumArray = this.computerNum.split('');
+      let userNumArray = userInputNumbers.split('');
+  
+      console.log(this.computerNum);
+      console.log(userInputNumbers);
+      computerNumArray.forEach(function(item, index) {
```

Comment:
```suggestion
      computerNumArray.forEach(function(item, index) {
```
혹시 map도있는데 forEach을 쓴 이유가 있으신가요?

Reply:
@hochan222 그것은 몰랐기 때문입니다...!! 하하..

Reply:
저도 잘은 모르지만 둘다 똑같은 역할을 하는데 forEach는 원본 array를 변화시키고 map은 새 배열을 반환합니다. 효율은 forEach가 빠른걸로 알고있습니다!